test/messaging.test.ts

```diff
@@ -0,0 +1,160 @@
+import { expect } from "chai";
+import hre, { ethers, starknet } from "hardhat";
+import { StarknetContract, HardhatUserConfig } from "hardhat/types";
+import config from "../hardhat.config";
+import { Contract, ContractFactory } from "ethers";
+import { Account } from "hardhat/types";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { getSelectorFromName } from "starknet/dist/utils/hash";
+
+const ASSET_SYMBOL = 10703902247957299200n;
+const ASSET_NAME = 4627187504670310400n;
+const ADDRESS_ACCOUNT = "0x0000000000000000000000000300000000000000";
+const BALANCE = 4412482n;
+const PUBLIC_KEY = 761466874539515783303110363281120649054760260892n
+const R = "0x6df1fd74c3334fa829febec3cb439bc8fa5e0702bf50260074355e21ac2d365a";
+const S = "0x000c65760f5886d4947a60caa12e6db8e43b11a7b063541dcc5e938cdd687ea3";
+const V = 0;
+
+describe("test contracts interaction", function () {
+  let l1Contract: Contract;
+  let l2Contract: StarknetContract;
+  let l1ContractFactory: ContractFactory;
+  let l2user: Account;
+  let selector: string;
+  let mockStarknetMessagingAddress: string;
+  let admin: SignerWithAddress;
+  let l1_user: SignerWithAddress;
+
+  before(async () => {
+    [admin, l1_user] = await hre.ethers.getSigners();
+    // const starkNetFake = await smock.fake(interface);
+    l2user = await starknet.deployAccount("OpenZeppelin");
+
+    const networkUrl =
+      (config as HardhatUserConfig).networks?.l1_testnet?.url ||
+      "http://localhost:8545";
+
+    mockStarknetMessagingAddress = (
+      await starknet.devnet.loadL1MessagingContract(networkUrl)
+    ).address;
+
+    const l2ContractFactory = await starknet.getContractFactory(
+      "contracts/starknet/proof_reserve"
+    );
+
+    l2Contract = await l2ContractFactory.deploy({
+      admin: l2user.starknetContract.address,
+      publisher: BigInt(761466874539515783303110363281120649054760260892n),
```

Comment:
could you add a constant for `761466874539515783303110363281120649054760260892n`?